server/model/monitor.js

```diff
@@ -809,6 +809,24 @@ class Monitor extends BeanModel {
                 bean.important = true;
 
                 if (Monitor.isImportantForNotification(isFirstBeat, previousBeat?.status, bean.status)) {
+                    // Change the notification msg if status becomes UP from DOWN
+                    // Adding msg for the downtime
+                    if (bean.status === UP) {
+                        const downAfterPreviousUpBeat = await R.findOne("heartbeat", " monitor_id = ? AND status = ? AND time > ( SELECT time from heartbeat WHERE monitor_id = ? AND status = ? ORDER BY time DESC ) ORDER BY time ASC", [
```

Comment:
> Given the current scaling issues, this is not ideal, but given that this query is only executed on notification if the bean is UP, this probably won't be an issue.

Yes, basically this query will only run when the service goes from DOWN to UP which is not a daily scenario for most of the service. Also, for cutting down the rows I am using filtering to get only those related to specific monitor by it's ID.
There is also no other way we can get the most recent UP beat without using the ordering of rows.

> Have you measured the impact of this query? How long does it take? (sqlite does not allow concurrent queries => long-running queries could cause downtime on bigger databases or slower IO (e.g. rasperry Pis using SD-Cards))
Could waiting here for longer be an issue?

This actually varies from system to system and also how big the table size is. In my system, there are only around 100 rows for a single monitor and it gives results in some milliseconds from DB. So, we can't really find out how it will impact other systems of different sizes.